### ISSUE TITLE
IncompatibleClassChangeError using 2.4.8 with Grails 2.2.4 and Java7

### ISSUE THREAD

**Issue by admiwi** [DiUS/pact-jvm#242]
When trying to instantiate `ConsumerInfo` in a I received this error:
```
java.lang.IncompatibleClassChangeError: the number of constructors during runtime and compile time for au.com.dius.pact.provider.PactVerification do not match. Expected -1 but got 2
```
A bit of research suggests this is an issue when using class compiled with Java6 in Groovy using Java7.

**Comment by uglyog** [comment]
Looks like there are 2 ways to fix this, use `@CompileStatic` or convert the class to Java. I'll see if I can fix it with the compile static annotation first.

**Comment by uglyog** [comment]
I converted the enum and exception classes to Java, and that resolved the issue. You may come across this again with other classes. If it occurs again, let me know.

**Comment by uglyog** [comment]
This has been released with versions 3.2.7 and 2.4.9
